core/ir/aarch64/instr_create.h

```diff
@@ -2565,5 +2565,136 @@ enum {
  */
 #define INSTR_CREATE_fmov_scalar_imm(dc, Rd, f) instr_create_1dst_1src(dc, OP_fmov, Rd, f)
 
+/**
+ * Creates a CCMN (Conditional Compare Negative) instruction.
+ * \param dc      The void * dcontext used to allocate memory for the instr_t.
+ * \param Rn      The source register.
+ * \param Op      Either a 5-bit immediate (use opnd_create_immed_uint(val, OPSZ_5b)
```

Comment:
Does this really need to be created with the `OPSZ_5b` size?  Could we change that and have the encoder handle any declared size?  Unless there are multiple encoding variants that differ in immediate sizes, the encoder shouldn't care so long as the value fits in 5 bits?  It would be best for usability to not require anyone to specify all these precise bit counts.  If a bit count does have to specified, there should be a named constant like `OPSZ_ccmn` or sthg like x86 has for `OPSZ_xlat`, `OPSZ_lgdt`, etc., and in addition maybe this macro should take just the integer and supply the size or set the size of the opnd before passing it on or sthg.

Reply:
For these instructions we need an unsigned integer immediate operand. The existing API for creating such operands (`opnd_create_immed_uint`) requires to specify an `opnd_size_t` value. How do you suggest to create an operand without using `opnd_create_immed_uint`? Or do you suggest to change `opnd_create_immed_uint`?

Reply:
Allow any size.  Or additionally maybe add `opnd_create_immed_uint_unsized` and internally it sets OPSZ_4 or sthg and the a64 encoder ignores the size since it has no encoding decisions based on size.

Reply:
I thought it is helpful to give an example of how to create a correct operand. It would be better if this macro took just an integer value, but most of `INSTR_CREATE_` macros accept operands. I can remove `OPSZ_5b` from the doc string.

Reply:
> I can remove OPSZ_5b from the doc string.

SGTM.  The other suggestions (like an _unsized version) are separate/larger scope from this PR.  I'm pretty sure the a64 encoder today ignores the size completely, so the user could pass OPSZ_NA or sthg.